src/routes/news/likes.ts

```diff
@@ -1,51 +1,43 @@
-import { ctxAccessGuard } from "$lib/utils/access";
+import { withAccess } from "$lib/utils/access";
 import apiNames from "$lib/utils/apiNames";
 import prisma from "$lib/utils/prisma";
-import { Prisma } from "@prisma/client";
 import { fail, type RequestEvent } from "@sveltejs/kit";
+import { message, superValidate } from "sveltekit-superforms/server";
+import { z } from "zod";
 
-export const modifyLikes = async (
-  { request, locals }: RequestEvent<Record<string, string>, string>,
-  shouldLike: boolean
-) => {
-  const session = await locals.getSession();
-  if (!session?.user?.student_id) {
-    return fail(401, { error: "Not logged in" });
-  }
-  await ctxAccessGuard(apiNames.NEWS.LIKE, session?.user);
-  const formData = await request.formData();
-  const articleId = formData.get("articleId");
-  if (!articleId) {
-    return fail(400, { error: "No article ID provided" });
-  }
-  const likers: Prisma.MemberUpdateManyWithoutLikedArticlesNestedInput = shouldLike
-    ? {
-        connect: {
-          studentId: session.user.student_id,
-        },
-      }
-    : {
-        disconnect: {
-          studentId: session.user.student_id,
-        },
-      };
-  try {
-    await prisma.article.update({
-      where: { id: String(articleId) },
-      data: {
-        likers,
-      },
-      select: {
-        id: true,
+export const likeSchema = z.object({
+  articleId: z.string(),
+});
+export type LikeSchema = typeof likeSchema;
+
+export const likesAction =
+  (shouldLike: boolean) =>
+  async ({ request, locals }: RequestEvent<Record<string, string>, string>) => {
+    const form = await superValidate(request, likeSchema);
+    if (!form.valid) return fail(400, { form });
+    const session = await locals.getSession();
+    return withAccess(
+      apiNames.NEWS.LIKE,
+      session?.user,
+      async () => {
+        await prisma.article.update({
+          where: { id: form.data.articleId },
+          data: {
+            likers: {
+              [shouldLike ? "connect" : "disconnect"]: {
+                studentId: session?.user?.student_id,
+              },
+            },
+          },
+          select: {
+            id: true,
+          },
+        });
+        return message(form, {
+          message: `${shouldLike ? "Gillat" : "Slutat gilla"} artikel`,
```

Comment:
Just include `artikel` in both of the strings, makes extracting them into translations so much easier.